solr/core/src/java/org/apache/solr/handler/admin/SplitOp.java

```diff
@@ -51,28 +52,41 @@
 import org.apache.solr.search.SolrIndexSearcher;
 import org.apache.solr.update.SolrIndexSplitter;
 import org.apache.solr.update.SplitIndexCommand;
+import org.apache.solr.update.UpdateHandler;
 import org.apache.solr.util.RTimer;
 import org.apache.solr.util.RefCounted;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * CoreAdminOp implementation for shard splits. This request is enqueued when {@link SplitShardCmd}
+ * is processed by the Overseer.
```

Comment:
Is "enqueue" the right word -- isn't it just a direct HTTP call, or is there actually a message in a queue?

I'd drop "is processed by the Overseer".

Reply:
message in a queue

Reply:
Which?  Can you substantiate this please?

Reply:
OverseerCollectionMessageHandler actually executes the message but while executing the message, the states are enqueued before pushing to ZK.

Reply:
I think it needs to be re-phrased. `This request is executed when SplitShardCmd is processed.`

Reply:
> OverseerCollectionMessageHandler actually executes the message but while executing the message, the states are enqueued before pushing to ZK.

OCMH processes SplitShardCmd from a ZK queue.  SplitShardCmd, in turn, makes an HTTP call to a specific node to invoke SplitOp.  That isn't a queue; it's HTTP.  Am I wrong or missing something?

Any way, I definitely agree with the simplified explanation.

Reply:
yes, in the end it's a call.